### PR TITLE
fix(cherow): fixes escaped yield & await identifiers

### PR DESCRIPTION
**Pull request by KFlash**

### CODE REVIEW

**Comment by KFlash** [comment]
@fkleuver Any thoughts?

File: packages/cherow/test/lexer/identifier_edge.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import { Token } from '../../src/token';
+
+// Both Await and Yield is valid as identifiers in sloppy mode,
+// so we are testing their escaped version in strict mode / module code.
+describe('Lexer - Yield and await as escaped identifiers', () => {
+
+  function pass(name: string, opts: any) {
+      function test(name: string, context: Context) {
+          it(name, () => {
+              const state = new State(opts.source, undefined, undefined);
```
Comment (g-plane) [inline packages/cherow/test/lexer/identifier_edge.ts]: Shorthand: `new State(opts.source)`

Comment (KFlash) [reply packages/cherow/test/lexer/identifier_edge.ts]: can't do it unless state is refactored and a `?` or ` | void` is added. But not worth it. This is just for testing purposes

Comment (g-plane) [reply packages/cherow/test/lexer/identifier_edge.ts]: I recommend to use `?`

**Comment by g-plane** [comment]
This PR should be a bug fix, so I changed the commit message.

**Comment by KFlash** [comment]
Merging this because I added a few other things. This can be fixed later if needed.

**Pull request merged**